src/algebra/direct_sum/ring.lean

```diff
@@ -374,7 +375,7 @@ variables [Π i, add_comm_monoid (A i)] [add_monoid ι] [gsemiring A]
 
 /-- The `semiring` structure derived from `gsemiring A`. -/
 instance grade_zero.semiring : semiring (A 0) :=
-function.injective.semiring (of A 0) dfinsupp.single_injective
+function.injective.semiring' (of A 0) dfinsupp.single_injective
```

Comment:
Given this is the only use of `semiring'`, I'd be tempted to just inline it here. I've been meaning to refactor this `grade_zero` stuff anyway, and if you inline it here then I can add a nicer instance later.

Reply:
Agreed.  There's also a potential nat.cast/int.cast diamond here since the graded structures don't contain cast operations yet.